.github/workflows/release.yml

```
@@ -349,6 +349,47 @@ jobs:
             const { owner, repo } = context.repo
             await github.rest.repos.deleteRelease({ owner, repo, release_id: ${{ needs.set-release-id.outputs.release_id }} })
 
+  pyk-publish:
+    name: 'Publish pyk'
+    runs-on: [self-hosted, linux, flyweight]
+    environment: production
+    permissions:
+      id-token: write
+    needs: [set-release-id, source-tarball]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
```

Nitpick:

```suggestion
      - name: Install Python
```

For consistency with the next step and https://github.com/runtimeverification/k/blob/f373630f28653b60b4079cce2242e3bab90ef42c/.github/actions/setup-pyk-env/action.yml#L12